src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/StraxCoinviewRule.cs

```diff
@@ -106,5 +123,32 @@ protected override void AllowSpend(TxOut prevOut, Transaction tx)
 
             // Otherwise allow the spend (do nothing).
         }
+
+        /// <inheritdoc />
+        public override async Task RunAsync(RuleContext context)
+        {
+            if (this.logic != null && context.ValidationContext.BlockToValidate.Header is PosBlockHeader posHeader && posHeader.HasSmartContractFields)
+                await this.logic.RunAsync(base.RunAsync, context);
```

Comment:
ConfigureAwait(false)

Reply:
That would differ from the legacy call in `SmartContractPoACoinviewRule`.

Reply:
I believe we have adding `ConfigureAwait(false)` in code policy
Unless there is justification why task should run on the same thread `ConfigureAwait(false)` should be used everywhere in our code